src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ListingGenerationController.java

```diff
@@ -321,8 +321,23 @@ public ResponseEntity<ListingCallbackResponse> generateHearingDocument(
 
         var listingData = listingRequest.getCaseDetails().getCaseData();
         var caseTypeId = listingRequest.getCaseDetails().getCaseTypeId();
-
-        return getResponseEntity(listingData, caseTypeId, userToken);
+        List<String> errorsList = new ArrayList<>();
+        boolean invalidCharsExist = listingService.checkInvalidCharsForAllParties(
+                listingRequest.getCaseDetails(), userToken, errorsList);
+        if (hasListings(listingData) && !invalidCharsExist) {
```

Comment:
is this if statement correct?
If you have listings but invalid chars exist then the user will also see the message for "No cases with hearings have been found". Is this what is intended?

Reply:
Corrected, "No cases with hearings have been found" error will only be added when no listing exists.